src/Logger-Arduino_AllPins/main.py

```diff
@@ -69,30 +71,36 @@ def __init__(self) -> None:
         self.plottype = [True for x in self.variables]  # True to plot data
         self.savetype = [True for x in self.variables]  # True to save data
 
+        # Communication Parameter
         self.port_manager = True
         self.port_types = ["COM"]
         self.port_properties = {
             "timeout": 3,
             "EOL": "\n",
             "baudrate": 115200,
         }
+        self.port_str: str = ""
+        self.driver_name: str = ""
 
+        # Device Parameter
         self.max_voltage = 5.0
-
+        self.resolution: int = 4096  # number of steps for 10 bit resolution
```

Comment:
4096 would correspond to 12bit. Furthermore I think one needs to subtract 1, to be consistent with how self.resolution is calculated in get_GUIparameter